### ISSUE TITLE
ConvTranspose2d 19.6x slower than cuDNN on MI355X for VAE decoder shapes

### ISSUE THREAD

**Issue by sunway513** [ROCmSoftwarePlatform/MIOpen#3952]
## Summary
`torch.nn.ConvTranspose2d` on MI355X (gfx950, MIOpen) is 19.6x slower than cuDNN on B300 for VAE decoder upsample shapes used in diffusion models (FLUX, SD3.5, SDXL).

## Measurements

| Shape | MI355X (MIOpen) | B300 (cuDNN 9.19) | Ratio |
|---|---|---|---|
| Cin=512, Cout=256, k=3, s=2, H=128, W=128 | **1.349ms / 28.7 TFLOPS** | 0.069ms / 562.7 TFLOPS | **19.6x slower** |

Tested with PyTorch 2.11, BF16, `cudnn.benchmark=True` on both sides.

## Context
This shape is the VAE decoder upsample layer present in ALL diffusion models (FLUX, SD3.5, SDXL, Sana, HunyuanVideo, Wan2.1, etc.). VAE decode is 60-70% Conv2D and is often the e2e latency bottleneck for image generation.

Standard Conv2d (non-transposed) on MI355X is competitive — only 0.78x vs cuDNN. The transposed convolution path appears severely underoptimized.

## Reproduce
```python
import torch, time
conv_t = torch.nn.ConvTranspose2d(512, 256, 3, stride=2, padding=1, output_padding=1, bias=False).cuda().bfloat16()
x = torch.randn(1, 512, 128, 128, dtype=torch.bfloat16, device='cuda')
[conv_t(x) for _ in range(100)]
torch.cuda.synchronize(); t=time.perf_counter()
[conv_t(x) for _ in range(200)]
torch.cuda.synchronize(); ms=(time.perf_counter()-t)/200*1000
flops = 2*1*512*256*256*256*3*3  # approximate
print(f'{flops/ms*1e-9:.1f} TFLOPS ({ms:.3f}ms)')
```

## Impact
Blocks competitive VAE decode performance on MI355X for all diffusion model inference.

cc @sunway513

**Comment by sunway513** [comment]
## Correction: 19.6x was a benchmark bug — actual gap is 2-3x

Re-measured with correct padding/stride settings:

| Shape | MI355X (MIOpen) | B300 (cuDNN 9.19) | Ratio |
|---|---|---|---|
| 512->256, k=3, s=2, 128x128 | 851 TFLOPS (0.18ms) | 1847 TFLOPS (0.08ms) | **0.46x** |
| 256->128, k=3, s=2, 256x256 | 461 TFLOPS (0.34ms) | 1455 TFLOPS (0.11ms) | **0.32x** |
| 512->512, k=3, s=2, 128x128 | 1222 TFLOPS (0.25ms) | 2731 TFLOPS (0.11ms) | **0.45x** |
| 128->128, k=3, s=2, 512x512 | 630 TFLOPS (0.49ms) | 913 TFLOPS (0.34ms) | **0.69x** |

The original 19.6x claim was from an incorrectly constructed ConvTranspose2d benchmark. The real gap is 2-3x, which is still significant for VAE decoder performance but not catastrophic.

For context, standard Conv2d (forward) gap is 0.78x, so ConvTranspose2d at 0.32-0.69x is worse than forward convolution — there may still be optimization opportunity in MIOpen's transposed convolution path.

**Comment by jychoi-hpc** [comment]
I want to follow this issue. I think I am having this issue too. I am using MI250X.
